lib/isochrone-tile.js

```diff
@@ -1,42 +1,30 @@
+// @flow
+import type {ImageData} from './types'
+
 /**
- * Fill ImageData object to be used on Canvas.
- * colorScheme is a function that takes a number of minutes and returns an rgba array of the color for that pixel
+ * Fill ImageData object to be used on Canvas. colorScheme is a function that
+ * takes a number of minutes and returns an rgba array of the color for that
+ * pixel.
  */
-
-import {hsl, rgb} from 'd3-color'
-import slice from 'lodash/slice'
-
-const colorTable = new Uint8Array(255 * 4)
-
-for (let i = 0; i < 254; i++) {
-  const val = Math.min(i, 120)
-  const hue = 45 * (val / 15 | 0) // 15 minutes of travel time represents 45 degrees on the color wheel
-  const saturation = (val % 15 + 5) / 20 // saturation represents gradation within a 15-minute period
-  const luminosity = (val % 15 + 5) / 20
-  const color = rgb(hsl(hue, saturation, luminosity))
-
-  colorTable[val * 4] = color.r
-  colorTable[val * 4 + 1] = color.g
-  colorTable[val * 4 + 2] = color.b
-  colorTable[val * 4 + 3] = 100
-}
-
-function defaultColorScheme (val) {
-  return slice(colorTable, val * 4, val * 4 + 4)
-}
-
-// 255 is preinitialized to 0, 0, 0, 0
-
-export default function isochroneTile (imageData, {
-  colorScheme = defaultColorScheme,
+export default function isochroneTile (imageData: ImageData, {
+  colorScheme,
   height,
   scaleFactor,
   surface,
   width,
   xoffset,
   yoffset
-}) {
-  // compiler should avoid overflow checks for xp and yp because of the < 256 condition, but prevent it from checking for pixel overflow with | 0
+}: {
+  colorScheme: Function,
```

Comment:
[The Function type is unsafe and should be avoided](https://flow.org/en/docs/types/functions/#toc-function-type).